allennlp/modules/matrix_attention/cosine_matrix_attention.py

```diff
@@ -7,20 +7,17 @@
 from overrides import overrides
 
 from allennlp.common import Params
-from allennlp.modules.matrix_attention import MatrixAttention
+from allennlp.modules.matrix_attention.matrix_attention import MatrixAttention
 
 
 @MatrixAttention.register("cosine")
 class CosineMatrixAttention(MatrixAttention):
 
-    def __init__(self) -> None:
-        super(CosineMatrixAttention, self).__init__()
-
     @overrides
     def forward(self, matrix_1: torch.Tensor, matrix_2: torch.Tensor) -> torch.Tensor:
         # pylint: disable=arguments-differ
         return F.cosine_similarity(matrix_1, matrix_1)
 
-
+    @classmethod
     def from_params(cls, params: Params):
         CosineMatrixAttention()
```

Comment:
It's important to call `params.assert_empty(cls.__name__)` before returning the object here (and in the corresponding methods for the other classes).  This helps to catch errors in the configuration file.  You're also missing a return statement.